astropy/time/formats.py

```diff
@@ -54,23 +55,64 @@ def _regexify_subfmts(subfmts):
     new_subfmts = []
     for subfmt_tuple in subfmts:
         subfmt_in = subfmt_tuple[1]
-        for strptime_code, regex in (('%Y', r'(?P<year>\d\d\d\d)'),
-                                     ('%m', r'(?P<mon>\d{1,2})'),
-                                     ('%d', r'(?P<mday>\d{1,2})'),
-                                     ('%H', r'(?P<hour>\d{1,2})'),
-                                     ('%M', r'(?P<min>\d{1,2})'),
-                                     ('%S', r'(?P<sec>\d{1,2})')):
-            subfmt_in = subfmt_in.replace(strptime_code, regex)
-
-        if '%' not in subfmt_in:
-            subfmt_tuple = (subfmt_tuple[0],
-                            re.compile(subfmt_in + '$'),
-                            subfmt_tuple[2])
+        if isinstance(subfmt_in, str):
+            for strptime_code, regex in (('%Y', r'(?P<year>\d\d\d\d)'),
+                                         ('%m', r'(?P<mon>\d{1,2})'),
+                                         ('%d', r'(?P<mday>\d{1,2})'),
+                                         ('%H', r'(?P<hour>\d{1,2})'),
+                                         ('%M', r'(?P<min>\d{1,2})'),
+                                         ('%S', r'(?P<sec>\d{1,2})')):
+                subfmt_in = subfmt_in.replace(strptime_code, regex)
+
+            if '%' not in subfmt_in:
+                subfmt_tuple = (subfmt_tuple[0],
+                                re.compile(subfmt_in + '$'),
+                                subfmt_tuple[2])
         new_subfmts.append(subfmt_tuple)
 
     return tuple(new_subfmts)
 
 
+def decimal_to_twoval1(val1, val2=None):
+    i, f = divmod(Decimal(val1), 1)
+    return float(i), float(f)
+
+
+decimal_to_twoval = np.vectorize(decimal_to_twoval1)
+
+
+def bytes_to_twoval1(val1, val2=None):
+    return decimal_to_twoval1(val1.decode('ascii'))
+
+
+bytes_to_twoval = np.vectorize(bytes_to_twoval1)
+
+
+def twoval_to_float128(val1, val2):
+    return val1.astype(np.float128) + val2.astype(np.float128)
+
+
+def twoval_to_decimal1(val1, val2):
```

Comment:
Perhaps it is important to beware of the interaction between the precision specified in the object and the precision with which Decimal objects are represented - the user could have an unpleasant surprise if the Decimal precision were not sufficient. There is a context manager that can set Decimal precision.

This is particularly an issue with the string conversions, where users may not realize that they are going through Decimal.

Some tests with deviant Decimal contexts would be a good idea.